node/src/Commands.ts

```diff
@@ -1089,34 +1089,45 @@ export function createZRank(
     return createCommand(RequestType.ZRank, args);
 }
 
-export type StreamTrimOptions = (
-    | {
-          /**
-           * Trim the stream according to entry ID.
-           * Equivalent to `MINID` in the Redis API.
-           */
-          method: "minid";
-          threshold: string;
-      }
-    | {
-          /**
-           * Trim the stream according to length.
-           * Equivalent to `MAXLEN` in the Redis API.
-           */
-          method: "maxlen";
-          threshold: number;
-      }
-) & {
+export type StreamTrimExactOptions = {
     /**
-     * If `true`, the stream will be trimmed exactly. Equivalent to `=` in the Redis API. Otherwise the stream will be trimmed in a near-exact manner, which is more efficient, equivalent to `~` in the Redis API.
+     * If `true`, the stream will be trimmed exactly. Equivalent to `=` in the Redis API.
+     * Otherwise, the stream will be trimmed in a near-exact manner, which is more efficient. Equivalent to
+     * `~` in the Redis API.
      */
-    exact: boolean;
+    exact?: boolean;
+};
+
+export type StreamTrimLimitOptions = {
     /**
      * If set, sets the maximal amount of entries that will be deleted.
      */
-    limit?: number;
+    exact: false;
+    limit: number;
 };
 
+export type StreamTrimMinIdOptions = {
+    /**
+     * Trim the stream according to entry ID.
+     * Equivalent to `MINID` in the Redis API.
+     */
+    method: "minid";
+    threshold: string;
+} & (StreamTrimExactOptions | StreamTrimLimitOptions);
```

Comment:
This combination does not allow to do
```
let options: StreamTrimMinIdOptions = {exact: true, threshold: "_"};
```
I don't know why, but you probably need to fix it and prove that it works in tests